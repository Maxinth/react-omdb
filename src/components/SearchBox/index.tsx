// import MovieList from "../MovieList";
import { Box, InputBox, Label } from "./searchBox.styled";
import React from "react";
interface Props {
  handleSearch: (id: string) => void;
  val: string;
}
const SearchBox = ({ handleSearch, val }: Props) => {
  return (
    <Box>
      <Label htmlFor="search">Search</Label>
      <InputBox
        type="text"
        value={val}
        id="search"
        onChange={(e) => handleSearch(e.target.value)}
      />
    </Box>
  );
};

export default SearchBox;
