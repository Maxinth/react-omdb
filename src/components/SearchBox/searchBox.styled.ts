import styled from "styled-components";

const Section = styled.section`
  background-color: #e5e5e5;
  width: 100%;
`;
const Box = styled.div`
  margin-top: 90px;
  /* width: 51px; */
  width: 100%;
  height: 21px;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  padding-left: 28px;
  padding-right: 27px;

  /* width: 100%; */

  @media (min-width: 834px) {
    /* position: static; */
    /* width: 680px; */
    height: 54px;
    left: 0px;
    top: 35px;
    padding-right: 77px;
    padding-left: 77px;
  }

  @media (min-width: 1440px) {
    padding-right: 57px;
  }
`;

const Label = styled.label`
  font-family: "DM Sans", sans-serif;
  font-size: 16px;
  font-style: normal;
  font-weight: 400;
  line-height: 21px;
  letter-spacing: 0em;
  text-align: left;
  align-self: flex-start;
`;
const InputBox = styled.input`
  width: 100%;
  min-height: 34px;
  border: 1px solid #000000;
  margin: 4px 0px;
  padding: 0 1rem;

  @media (min-width: 834px) {
    min-height: 54px;
  }
`;
export { Box, InputBox, Label, Section };
