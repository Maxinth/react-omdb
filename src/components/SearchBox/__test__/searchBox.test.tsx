import SearchBox from "..";
import renderer from "react-test-renderer";
import React from "react";

const mockProps = {
  handleSearch: jest.fn(),
  val: "80j",
};
it("renders a snapshot of <SearchBox /> with mockProp", () => {
  const searchBox = renderer.create(<SearchBox {...mockProps} />).toJSON();
  expect(searchBox).toMatchSnapshot();
});
