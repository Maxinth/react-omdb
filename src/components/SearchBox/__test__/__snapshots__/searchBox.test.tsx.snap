// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot of <SearchBox /> with mockProp 1`] = `
<div
  className="sc-gsDKAQ eMXzRO"
>
  <label
    className="sc-dkPtRN dCFilw"
    htmlFor="search"
  >
    Search
  </label>
  <input
    className="sc-hKwDye XTisZ"
    id="search"
    onChange={[Function]}
    type="text"
    value="80j"
  />
</div>
`;
